src/rust/en.mangafox/src/parser.rs

```diff
@@ -0,0 +1,362 @@
+use aidoku::{
+	error::Result,
+	helpers::substring::Substring,
+	prelude::*,
+	std::{html::Node, String, Vec},
+	Chapter, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaStatus,
+	MangaViewer, Page,
+};
+
+use crate::unpacker;
+
+extern crate alloc;
+use alloc::string::ToString;
+
+pub fn parse_directory(html: Node) -> Result<MangaPageResult> {
+	let mut result: Vec<Manga> = Vec::new();
+	let has_more: bool = !is_last_page(html.clone());
+
+	for page in html.select("ul.line li").array() {
+		let obj = page.as_node().expect("html array not an array of nodes");
+
+		let id = obj
+			.select("a")
+			.attr("href")
+			.read()
+			.replace("/manga/", "")
+			.replace('/', "");
+		let title = obj.select("a").attr("title").read();
+		let cover = obj.select("a img").attr("src").read();
+
+		result.push(Manga {
+			id,
+			cover,
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: String::new(),
+			categories: Vec::new(),
```

Comment:
use `..Default::default()` at the end instead of making these blank strings and stuff